proxy/plugins/authorize/authorize.go

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package authorize
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+)
+
+var authorizePlugins = map[string]AuthPlugin{}
+var authorMap = map[string]Author{}
+
+//AuthPlugin is a function of config file and turns the Author interface
+type AuthPlugin = func(configFile string) (Author, error)
+
+//Author provide Authorize and Identify interface to auth authorization
+type Author interface {
+	Authorize(username string, password string) error
+	Identify(req *http.Request) error
+}
+
+//RegisterAuthPlugin support the authorizePlugins register
+func RegisterAuthPlugin(kind string, pluginFunc AuthPlugin) {
+	_, ok := authorizePlugins[kind]
+	if ok {
+		log.Printf("authorize is already exit, name = %s", kind)
```

Comment:
Why not use openlogging as before? What is the difference between log and openlogging?